### PR TITLE
Fix snapshot build and depend on OpenSearch 1.1.

### PR DESCRIPTION
**Pull request by dblock**
Signed-off-by: dblock <dblock@amazon.com>

### Description

Fixes snapshot build, defaults to it in CI, takes a dependency and inherits version from OpenSearch 1.1.
 
### Issues Resolved

Closes https://github.com/opensearch-project/dashboards-notebooks/issues/60
 
### Check List
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: opensearch-notebooks/build.gradle [hunk 1]
```diff
@@ -29,7 +29,10 @@ import java.util.concurrent.Callable
 
 buildscript {
     ext {
-        opensearch_version = System.getProperty("opensearch.version", "1.0.0")
+        opensearch_version = System.getProperty("opensearch.version", "1.1.0-SNAPSHOT")
+        // 1.0.0 -> 1.0.0.0, and 1.0.0-SNAPSHOT -> 1.0.0.0-SNAPSHOT
+        opensearch_build = opensearch_version.replaceAll(/(\.\d)([^\d]*)$/, '$1.0$2')
```
Comment (kartg) [inline opensearch-notebooks/build.gradle:34]: Why do we want to implicitly morph the incoming version with an extra `.0` ?

**Review by kartg — APPROVED** [review]

File: opensearch-notebooks/build.gradle [hunk 2]
```diff
@@ -29,7 +29,10 @@ import java.util.concurrent.Callable
 
 buildscript {
     ext {
-        opensearch_version = System.getProperty("opensearch.version", "1.0.0")
+        opensearch_version = System.getProperty("opensearch.version", "1.1.0-SNAPSHOT")
```
[the remark this answers is not in the document]
Comment (dblock) [reply opensearch-notebooks/build.gradle:32]: We could. But we do this in so many plugins, so I'd prefer to do this as a separate issue/PR, and standardize across the set.

File: opensearch-notebooks/build.gradle (hunk 1 above)
Comment (dblock) [reply opensearch-notebooks/build.gradle:34]: OpenSearch is a 3-digit version, so you start with 1.1.0 or 1.1.0-SNAPSHOT, but common utils and plugins are 4-digit, so 1.1.0.0 and 1.1.0.0-SNAPSHOT. We have https://github.com/opensearch-project/OpenSearch/issues/1093 open to stop doing that.

Comment (dblock) [reply opensearch-notebooks/build.gradle:34]: This is just a clever(er) way of doing `opensearch_version - '-SNAPSHOT' + '.0' + ('-SNAPSHOT' if opensearch_version.contains('-SNAPSHOT') else '').

**Review by joshuali925 — APPROVED** [review]
thanks!

**Pull request merged**